### PR TITLE
BShisen [recipe]

### PR DESCRIPTION
**Pull request by bach5000**
**This recipe is untested...**
Some information is probably not correct. This will be fixed later.
But anyone is welcome to make/give reviews 😄 

I'll test and fix this recipe once I get home.

### CODE REVIEW

**Comment by scottmc** [comment]
on the right hand side there are labels you can add to pending pull requests.  I have added the wip one, this let's us see the current state of pending requests at a glance.

File: haiku-apps/bshisen/bshisen-1.2.recipe [hunk 1]
```diff
@@ -0,0 +1,42 @@
+SUMMARY="BeOS tile matching game"
+DESCRIPTION="BShisen is a Majong style game made for Haiku."
+HOMEPAGE="https://github.com/HaikuArchives/BShisen"
+COPYRIGHT="1998-1999 Kelvin W Sherlock"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="1e86e4daec9f48d9aa9642227f962159cfe3a733"
+SOURCE_URI="https://github.com/HaikuArchives/BShisen/archive/$srcGitRev.tar.gz"
+#CHECKSUM_SHA256="e0b9844cebb6ed2cf42f93e592fe2e4ce0b801457b31cbac7ff7f5b5d493f76e"
```
Comment (fbrosson) [inline haiku-apps/bshisen/bshisen-1.2.recipe]:
```diff
-#CHECKSUM_SHA256="e0b9844cebb6ed2cf42f93e592fe2e4ce0b801457b31cbac7ff7f5b5d493f76e"
+CHECKSUM_SHA256="263343a257a313358fee3b8bf3448cb50ab15f8e4c8b0ecd7e12b2e408822ba6"
```

BTW, the game is working perfectly on x86_gcc2 :)

**Comment by scottmc** [comment]
While you are at it...  can you add a screenshot for it on HaikuArchives?  and put it into the readme.md?  Make it a good one, perhaps in middle of a game.

File: haiku-apps/bshisen/bshisen-1.2.recipe [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+SOURCE_FILENAME="BShisen-$srcGitRev.tar.gz"
+SOURCE_DIR="BShisen-$srcGitRev/source"
+
+ARCHITECTURES="?x86_gcc2 ?x86_64"
```
Comment (fbrosson) [inline haiku-apps/bshisen/bshisen-1.2.recipe]:
```diff
-ARCHITECTURES="?x86_gcc2 ?x86_64"
+ARCHITECTURES="x86_gcc2 !x86_64"
```
On x86_gcc2 the compilation succeeds for BitmapControl.cpp, Prefs.cpp, Tiles.cpp and ID_Win.cpp but fails on main.cpp:
```
g++ -c main.cpp -iquote./  -iquote./   -O1   -Woverloaded-virtual -funsigned-bitfields -Wwrite-strings  -o "objects/main.o"
In file included from Shisen.h:9:0,
                 from main.cpp:14:
YLanguageClass.h:79:7: error: 'map' does not name a type
       map <const char*,const char*,ltstr>LangList;
```
So you can probably just mark x86_64 as broken.

File: haiku-apps/bshisen/bshisen-1.2.recipe (hunk 1 above)
Comment (bach5000) [reply haiku-apps/bshisen/bshisen-1.2.recipe]: I intended to wait until after school, but thanks for testing it for me @fbrosson :)

File: haiku-apps/bshisen/bshisen-1.2.recipe [hunk 3]
```diff
@@ -0,0 +1,42 @@
+SUMMARY="BeOS tile matching game"
+DESCRIPTION="BShisen is a Majong style game made for Haiku."
+HOMEPAGE="https://github.com/HaikuArchives/BShisen"
+COPYRIGHT="1998-1999 Kelvin W Sherlock"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="1e86e4daec9f48d9aa9642227f962159cfe3a733"
+SOURCE_URI="https://github.com/HaikuArchives/BShisen/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="263343a257a313358fee3b8bf3448cb50ab15f8e4c8b0ecd7e12b2e408822ba6"
+SOURCE_FILENAME="BShisen-$srcGitRev.tar.gz"
```
Comment (fbrosson) [inline haiku-apps/bshisen/bshisen-1.2.recipe]:
```diff
-SOURCE_FILENAME="BShisen-$srcGitRev.tar.gz"
+SOURCE_FILENAME="BShisen-$portVersion-$srcGitRev.tar.gz"
```

**Comment by bach5000** [comment]
@fbrosson You're welcome 😄 
@scottmc I added the screenshot as a pull request here:
https://github.com/HaikuArchives/BShisen/pull/3

**Comment by pulkomandy** [comment]
On Wed, Jan 10, 2018 at 08:37:18PM +0000, fbrosson wrote:
> On x86_gcc2 the compilation succeeds for BitmapControl.cpp, Prefs.cpp, Tiles.cpp and ID_Win.cpp but fails on main.cpp:
> ```
> g++ -c main.cpp -iquote./  -iquote./   -O1   -Woverloaded-virtual -funsigned-bitfields -Wwrite-strings  -o "objects/main.o"
> In file included from Shisen.h:9:0,
>                  from main.cpp:14:
> YLanguageClass.h:79:7: error: 'map' does not name a type
>        map <const char*,const char*,ltstr>LangList;
> ```
> So you can probably just mark x86_64 as broken.

Or fix the build: just use std::map instead of just "map", I think.
But we should probably remove this class and use the Locale Kit instead?

-- 
Adrien.

**Comment by fbrosson** [comment]
Thanks @pulkomandy for the hint.

@bach5000, if/when yiu have some time, would you like to investigate this?
(If you wish to, you can even leave that to after the end of the GCI, in case you would like to claim a few more tasks before the end.)

Regarding the `std::map`, I've seen some in patches which were sent to upstream projects, but I can't remember which right now. Looking at haikuport's `git log -p` I found a few occurences of it in cmake, so I guess that if you browse cmake's sources you will find some and maybe figure out how to fix BShisen's build on x86_64.

**Pull request merged**